src/protocols/secure_channel/CASEServer.cpp

```diff
@@ -100,7 +99,39 @@ void CASEServer::Cleanup()
     ChipLogProgress(Inet, "CASE Server enabling CASE session setups");
     mExchangeManager->RegisterUnsolicitedMessageHandlerForType(Protocols::SecureChannel::MsgType::CASE_Sigma1, this);
 
+    //
+    // If we previously had a session to a peer, latch it its details. If we didn't, this will
+    // contain kInvalidPeerNodeId.
+    //
+    ScopedNodeId previouslyEstablishedPeer = GetSession().GetPeer();
+
     GetSession().Clear();
+
+    //
+    // Indicate to the underlying CASE session to prepare for session establishment requests coming its way. This will
+    // involve allocating a SecureSession that will be held until it's needed for the next CASE session handshake.
+    //
+    // Logically speaking, we're attempting to evict a session using details of the just-established session (to ensure
+    // we're evicting sessions from the right fabric if needed) and then transferring the just established session into that
+    // slot (and there-by free'ing up the slot for the next session attempt). However, this transfer isn't necessary - just
+    // evicting a session will ensure it is available for the next attempt.
+    //
+    VerifyOrDie(GetSession().PrepareForSessionEstablishment(
+                    *mSessionManager, mFabrics, mSessionResumptionStorage, this, previouslyEstablishedPeer,
+                    Optional<ReliableMessageProtocolConfig>::Value(GetLocalMRPConfig())) == CHIP_NO_ERROR);
```

Comment:
In the absence of actual eviction logic, it is indeed better to die then to have the CASE server fail as before if no session slots are available because that leaves nodes deaf.

However, even if this is temporary, this should be in the comments: we are purposefully dying here to prevent nodes from becoming unreachable, and we don't expect this to be necessary (or at least not expected) once eviction is implemented.